hls_vi/generate_stac_items.py

```diff
@@ -0,0 +1,333 @@
+    },
+    "NBR": {
+        "band": Band.create(
+            name="NBR",
+            common_name="NBR",
+        ),
+        "gsd": 30.0,
+    },
+    "NBR2": {
+        "band": Band.create(
+            name="NBR2",
+            common_name="NBR2",
+        ),
+        "gsd": 30.0,
+    },
+    "NDMI": {
+        "band": Band.create(
+            name="NDMI",
+            common_name="NDMI",
+        ),
+        "gsd": 30.0,
+    },
+    "NDWI": {
+        "band": Band.create(
+            name="NDWI",
+            common_name="NDWI",
+        ),
+        "gsd": 30.0,
+    },
+    "SAVI": {
+        "band": Band.create(
+            name="SAVI",
+            common_name="SAVI",
+        ),
+        "gsd": 30.0,
+    },
+    "TVI": {
+        "band": Band.create(
+            name="TVI",
+            common_name="TVI",
+        ),
+        "gsd": 30.0,
+    },
+}
+
+
+def get_geometry(granule):
+    """Function returns the geometry of the HLS_VI granule
+
+    Args:
+        granule (untangle.Element): HLS_VI Granule information from the xml file
+
+    Returns:
+        MultiPolygon: Coordinates information about the granule
+    """
+    multipolygon = []
+    for poly in granule.Spatial.HorizontalSpatialDomain.Geometry.GPolygon:
+        ring = []
+        for point in poly.Boundary.Point:
+            geojson_point = [
+                float(point.PointLongitude.cdata),
+                float(point.PointLatitude.cdata),
+            ]
+            ring.append(geojson_point)
+
+        closing_point = [
+            float(poly.Boundary.Point[0].PointLongitude.cdata),
+            float(poly.Boundary.Point[0].PointLatitude.cdata),
+        ]
+        ring.append(closing_point)
+        ringtuple = (ring,)
+        multipolygon.append(ringtuple)
+    geometry = MultiPolygon(multipolygon)
+    return geometry
+
+
+def process_common_metadata(item, granule):
+    """Function fetches and processes the general information from the granule metadata.
+     and updates the information in the STAC item.
+
+    Args:
+        item (PyStac item): STAC item created from the HLS_VI granule
+        granule (untangle.Element): HLS_VI Granule information from the xml file
+    """
+    start_datetime_str = granule.Temporal.RangeDateTime.BeginningDateTime.cdata
+    item_start_datetime = datetime.datetime.strptime(
+        start_datetime_str, "%Y-%m-%dT%H:%M:%S.%fZ"
+    )
+    end_datetime_str = granule.Temporal.RangeDateTime.EndingDateTime.cdata
+    item_end_datetime = datetime.datetime.strptime(
+        end_datetime_str, "%Y-%m-%dT%H:%M:%S.%fZ"
+    )
+    item.common_metadata.start_datetime = item_start_datetime
+    item.common_metadata.end_datetime = item_end_datetime
+    item.common_metadata.platform = granule.Platforms.Platform.ShortName.cdata.lower()
+    instrument = (
+        granule.Platforms.Platform.Instruments.Instrument.ShortName.cdata.lower()
+    )
+    if " " in instrument:
+        item_instrument = instrument.split()[1]
```

Comment:
What's this about? Please add a comment explaining the need for this.

Reply:
@madhuksridhar Could you help here as well? I think it is something related to the sensors but I am not sure.

Reply:
@chuckwondo and @SwordSaintLancelot 
Form the metadata,
ShortName for L30 granule: `OLI`
ShorName for S30 granule: `Sentinel-2 MSI`
looks like this was used as a check to exclude `Sentinel-2` from the ShortName string for S30 granules.